server/server_test.go

```diff
@@ -436,6 +436,49 @@ func TestServiceRegistryInfoWithNilTags(t *testing.T) {
 	test.Assert(t, drCount == 1)
 }
 
+func TestServiceRegistryInfoSkipListenAddr(t *testing.T) {
```

Comment:
This test only covers the path of setting the flag, another case covering the path of not setting this flag is also necessary.